ruby/src/IceRuby/Communicator.cpp

```diff
@@ -41,12 +41,24 @@ IceRuby_Communicator_mark(Ice::CommunicatorPtr* p)
 }
 
 extern "C" void
-IceRuby_Communicator_free(Ice::CommunicatorPtr* p)
+IceRuby_Communicator_free(void* p)
 {
-    assert(p);
-    delete p;
+    auto communicator = static_cast<Ice::CommunicatorPtr*>(p);
+    delete communicator;
 }
 
+static const rb_data_type_t IceRuby_CommunicatorType = {
+    .wrap_struct_name = "Ice::Communicator",
+    .function =
+        {
+            .dmark = IceRuby_Communicator_mark,
+            .dfree = IceRuby_Communicator_free,
+            .dsize = nullptr,
```

Comment:
We should do the same for `dsize` and `dcompact`. We are initializing one explicitly to nullptr and the other implicitly. Same for `.data` below.